src/instantsearch/widget.js

```diff
@@ -10,34 +10,89 @@ class AlgoliaPlacesWidget {
     }
 
     this.placesOptions = placesOptions;
+    this.placesAutocomplete = places(this.placesOptions);
+
+    this.state = {};
   }
-  init({ helper }) {
-    const placesAutocomplete = places(this.placesOptions);
 
+  init({ helper }) {
     helper
       .setQueryParameter('insideBoundingBox')
-      .setQueryParameter('aroundLatLng', this.defaultPosition);
+      .setQueryParameter(
+        'aroundLatLng',
+        this.state.position || this.defaultPosition
+      );
 
-    placesAutocomplete.on('change', opts => {
+    this.placesAutocomplete.on('change', opts => {
       const {
         suggestion: {
           latlng: { lat, lng },
+          value,
         },
       } = opts;
 
+      this.state.position = `${lat},${lng}`;
+      this.state.query = value;
```

Comment:
Rather than duplicate the source of truth for the values why doesn't read them directly from the helper & placesAutocomplete?

Reply:
This is not the query you're looking for. It's actually the value for searching for values in places, which is nowhere in the main search state.

Reply:
Yes but since you have access to the Places instance you can have access to it.

Reply:
Although you can simplify the internal state by removing the position information, you cannot make it completely disappear, as the `getVal` method from the placesAutocomplete instance will yield stale data.

the `placesAutocomplete.on('change', () => {})` is triggered before the update of the value of the input, and so is `getWidgetState`. So `placesAutocomplete.getVal` will yield the previous input value, instead of the new one.

Reply:
Ok thanks for the update!